### PR TITLE
Small fixes after testing

### PR DESCRIPTION
**Pull request by eduardonunesp**
* Force slot and blockNumber toString
* Converting addresses to checkSum addresses or Metamask will fail
* getCoinFromTxAsync using index 0 (length == 1)
* getting event deposit correctly

### CODE REVIEW

File: src/plasma-cash/user.ts [hunk 1]
```diff
@@ -83,7 +83,7 @@ export class User extends Entity {
   async depositETHAsync(amount: BN): Promise<IPlasmaCoin> {
     let currentBlock = await this.getCurrentBlockAsync()
     const tx = await this.web3.eth.sendTransaction({
-      from: this.ethAddress,
+      from: this.web3.utils.toChecksumAddress(this.ethAddress),
```
Comment (gakonst) [inline src/plasma-cash/user.ts]: Move the conversion to a checksumed address to the constructor

File: src/plasma-cash/ethereum-client.ts [hunk 2]
```diff
@@ -183,13 +183,15 @@ export class EthereumPlasmaClient {
...
+    const root = await this._plasmaContract.methods
+      .getBlockRoot(blockNumber.toString())
+      .call({ from })
     return root
   }
 
   async getPlasmaCoinAsync(params: { slot: BN; from: string }): Promise<IPlasmaCoin> {
     const { slot, from } = params
-    const coin = await this._plasmaContract.methods.getPlasmaCoin(slot).call({ from })
+    const coin = await this._plasmaContract.methods.getPlasmaCoin(slot.toString()).call({ from })
```
Comment (gakonst) [inline src/plasma-cash/ethereum-client.ts]: Is this still a problem?

Comment (eduardonunesp) [reply src/plasma-cash/ethereum-client.ts]: Yes, still a problem

File: src/plasma-cash/user.ts [hunk 3]
```diff
@@ -100,7 +100,8 @@ export class User extends Entity {
     const tx = await token.methods
       .safeTransferFrom(this.ethAddress, this.plasmaCashContract._address, uid.toString())
       .send({ from: this.ethAddress, gas: this._defaultGas })
-    const coin = await this.getCoinFromTxAsync(tx.transactionHash, 1) // 2 events, transferred & deposited, we want the 2nd one
+    const coin = await this.getCoinFromTxAsync(tx.transactionHash) // 2 events, transferred & deposited, we want the 2nd one
```
Comment (enlight) [inline src/plasma-cash/user.ts]: `getCoinFromTxAsync` expects a tx, not just the hash.

**Pull request merged**